src/context.cc

```diff
@@ -274,6 +274,9 @@ void ContextBase::onCreate(uint32_t parent_context_id) {
     DeferAfterCallActions actions(this);
     wasm_->on_context_create_(this, id_, parent_context_id);
   }
+  // NB: If no on_context_create function is registered the in-VM SDK is responsible for
+  // managing any required in-VM state.
+  in_vm_context_created_ = true;
```

Comment:
Shouldn't we require `proxy_on_context_create` to be exported then? Right now, the contract is described in this comment, but it's not enforced not even checked anywhere.

Reply:
I don't feel that is necessary.  It is quite possible that some SDK subset will be stateless and not care about creating or destroying contexts.  Not making the calls would be more efficient then.

Reply:
(Ack. I'll get back to you on this "soon")

Reply:
Ping

Reply:
That is not what I was trying to say.  Instead I meant that if the plugin is stateless and does not require a context to be created at all then it could not export proxy_on_context_create and not create a context in the proxy_on_XXX calls at all.

Reply:
That said, I am fine with requiring this if you want.  I just thought it would be easier on SDK developers if there was a simple policy of not requiring functions to be exported and just ignoring the call if they  are not.

Reply:
Also It makes backward compatibility easier.

Reply:
But that's not at all what this code is doing. The code right now is silently masking the missing `proxy_on_context_create` callback and sets `in_vm_context_created_ = true`, letting other functions know that context was in fact created. As such, `proxy_on_http_XXX` callbacks will be called on context that was never created, leading to unclear behavior.

Like I've mentioned a while ago, if you want to make `proxy_on_context_create` optional, but only for stateless calls, then this should be enforced somewhere, and we could either always return `Continue` and bypass those calls, or simply reject Wasm modules that export those calls, but not `proxy_on_context_create`.

Reply:
enum InVmContext { NoRequired, Created, NotCreated }

Reply:
Better yet.  This code will be updated to match the ABI spec.  So it really doesn't matter what we decide here.  This is an intermediate solution until we update to match the spec.